### PR TITLE
feat: `into` variable declaration

### LINKED ISSUE #2427: Idea for more PRQLish table creation grammar
### What's up?

Currently, I think the only way to create tables is

```prql
let TableA = (
   from data
   ...
)
```

But it might be useful for a number of reasons to allow some pipeline operator for creating tables like this:
```
from data
...
let TableA = _
```

The point would be to allow components (table definition) to avoid nesting structures that basically block the ability to simply use concat/append as the base operator for constructing pipelines.

So someone might define some vanilla PRQL pipeline and you could then take that and simply extend it more more pipeline logic OR turn it into a table simply by + "let TableA" which sounds like not a big deal but removes a lot of complexity that leaks out of the components when you need to keep track of the type of the query object in order to combine with other query objects.

## UPDATE:

And to be clear, the intention is to enable the following
```
from data
...
let TableA = _

from data
...
join TableA [==something]
...
```

### PR DESCRIPTION
**Pull request by aljazerzen**
Based on #2575

Closes https://github.com/PRQL/prql/issues/2427

### CODE REVIEW

File: web/book/src/syntax/keywords.md [hunk 1]
```diff
@@ -2,10 +2,12 @@
 
 At the moment, PRQL uses only four keywords:
```
Comment (max-sixty) [inline web/book/src/syntax/keywords.md:3]:
```suggestion
At the moment, PRQL uses only six keywords:
```

**Comment by max-sixty** [comment]
Awesome! Thanks a lot for taking this further. 

I left it in a somewhat messy state — `FuncDef_` with the underscore, lots of things commented out, let me know if I can help clear it up.

**Comment by aljazerzen** [comment]
Don't worry about `FuncDef_`, this PR contains commits from #2575 and will be rebased after that is merged.

File: web/book/src/syntax/keywords.md (hunk 1 above)
Comment (max-sixty) [reply web/book/src/syntax/keywords.md:3]: Still six! Will change

**Comment by max-sixty** [comment]
Amazing, great work! I can write some docs...

**Pull request merged**